seastar/src/spawn.rs

```diff
@@ -32,6 +32,8 @@ pub async fn spawn<T, Ret: 'static>(future: T) -> Option<Ret>
 where
     T: Future<Output = Ret> + 'static,
 {
+    seastar::assert_runtime_is_running!();
```

Comment:
`Ensure safety of spawn`

Please squash this commit with `Add functions to spawn future on the local shard`.